### DISCUSSION TITLE
The requested module '@socket.io/component-emitter' does not provide an export named 'Emitter'

### DISCUSSION THREAD

**Discussion by professional-procrastinator** [socketio/socket.io#4210] [Q&A] [answered]
I know the title sounds confusing, and I'll try my best to explain the issue. 

So I created a next.js app, and installed socket.io-client.
I have a server running already, but I'm sure that it's not even related to the problem.

When I try to import socket.io-client, I get an error.

`import {io} from 'socket.io-client';`

`SyntaxError: The requested module '@socket.io/component-emitter' does not provide an export named 'Emitter' `

I literally don't have any idea why this happens. I've used socket.io with create-react-app and never ran into this issue. I've tried using older versions, like socket.io-client@3.1.2 and more. But every time, I get the exact same error.

I'm not even emitting events, or listening for an event, just importing socket.io-client. 

I'm new to next.js, and it's entirely possible that I'm doing something wrong. 
I'm using `socket.io-client@4.4.0` and `socket.io@4.4.0`. 

Should I try to include the bundle from a CDN, or are there any solutions? 
Thanks.

![image](https://user-images.githubusercontent.com/69335199/145389303-8813c376-2c5a-453d-9148-cbd43a7e3925.png)

**Comment by darrachequesne** [comment, marked as answer]
That's weird, the named export is here: https://github.com/socketio/emitter/blob/cd703fe28e5bf85ecf137b0e6422e2608c0eefbf/index.js#L6

Could you please provide a complete example reproducing the issue?

**Reply by professional-procrastinator** [reply]
Unfortunately, I think it's an issue with my project only. I tried to set up socket.io and next.js on https://codesandbox.io/, and it works nicely.
So, I don't really think this is a problem with socket.io. I eventually had to downgrade to an older version of socket.io-client, which worked, and in this case, it was version 3.0.0. 

I'm still really confused why this even happened.
I think something probably went wrong when I was trying to install the package.  Also, I **have used socket.io@4.4.0** with React once, and it worked flawlessly. I think the people who might see this discussion later should first make a new directory with their project code, and then install socket.io. Again, I'm not really sure why that should be the solution.

But thanks a lot for the help!

**Comment by game-geek** [comment]
Hello
I have a expo app with socket.io and socket.io-client and ran into the same issue.

Switching socket.io-client to version 4.0 solves the problem, but it's not the way to go.
After more debugging, I found out that just **uninstalling socket.io**, and only having socket.io-client solves the issue!
I don't know if it's a bug (maybe just my project) or if it's made on purpose that both packages don't work in the same project.

Also this bug only occurs on expo web, but not on expo android emulation.

Hope it helped somebody.
